New `otelcol.exporter.debug` component (#5867)
grafana/agent: docs/sources/flow/reference/components/otelcol.exporter.debug.md

---
aliases:
- /docs/grafana-cloud/agent/flow/reference/components/otelcol.exporter.debug/
- /docs/grafana-cloud/monitor-infrastructure/agent/flow/reference/components/otelcol.exporter.debug/
- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/reference/components/otelcol.exporter.debug/
- /docs/grafana-cloud/send-data/agent/flow/reference/components/otelcol.exporter.debug/
canonical: https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.exporter.debug/
description: Learn about otelcol.exporter.debug
labels:
  stage: experimental
title: otelcol.exporter.debug
---

# otelcol.exporter.debug

> `otelcol.exporter.debug` accepts telemetry data from other `otelcol` components
> and writes them to the console (stderr). The verbosity of the logs can also be controlled.

Reviewer: ```suggestion
`otelcol.exporter.debug` accepts telemetry data from other `otelcol` components and writes them to the console (stderr).
You can control the verbosity of the logs.
```

> > **NOTE**: `otelcol.exporter.debug` is a wrapper over the upstream
> > OpenTelemetry Collector `debug` exporter. Bug reports or feature requests will
> > be redirected to the upstream repository, if necessary.

Reviewer: ```suggestion
{{< admonition type="note" >}}
otelcol.exporter.debug` is a wrapper over the upstream OpenTelemetry Collector `debug` exporter.
If necessary, bug reports or feature requests are redirected to the upstream repository.
{{< /admonition >}}
```

> Multiple `otelcol.exporter.debug` components can be specified by giving them
> different labels.

Reviewer: ```suggestion
You can specify multiple `otelcol.exporter.debug` components by giving them different labels.
```

We can break on sentences vs arbitrary (80 column).

## Usage

```river
otelcol.exporter.debug "LABEL" { }
```

## Arguments

`otelcol.exporter.debug` supports the following arguments:

Name | Type | Description | Default | Required
---- | ---- | ----------- | ------- | --------

> `verbosity`           | `string` | Verbosity of the generated logs. | `"basic"` | no

Reviewer 2: ```suggestion
`verbosity`           | `string` | Verbosity of the generated logs. | `"normal"` | no
```
The default [upstream](https://github.com/open-telemetry/opentelemetry-collector/tree/v0.86.0/exporter/debugexporter) is "normal", so let's use the same one here.

`sampling_initial`    | `int`    | Number of messages initially logged each second. | `2` | no
`sampling_thereafter` | `int`    | Sampling rate after the initial messages are logged. | `500` | no

The `verbosity` argument must be one of `"basic"`, `"normal"`, or `"detailed"`.

## Exported fields

The following fields are exported and can be referenced by other components:

Name | Type | Description
---- | ---- | -----------
`input` | `otelcol.Consumer` | A value that other components can use to send telemetry data to.

`input` accepts `otelcol.Consumer` data for any telemetry signal (metrics,
logs, or traces).

## Component health

`otelcol.exporter.debug` is only reported as unhealthy if given an invalid
configuration.

## Debug information

`otelcol.exporter.debug` does not expose any component-specific debug
information.

## Example

> This example scrapes prometheus unix metrics and writes them to the console:

Reviewer: ```suggestion
This example scrapes Prometheus UNIX metrics and writes them to the console:
```
